transformer/transform.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/local/bin/python3.6
+import json
+import logging
+import sys
+import typing
+import uuid
+from collections import namedtuple
+
+import argparse
+
```

Comment:
Thinking of someone else coming in to maintain/evolve this transformer in the future ...
I think it would be helpful to have a comment block providing an overview of the transformation process, including how the many dicts in the input are associated and grouped into bundles.
This would really help in ramping-up on how the transformer works, as compared to reverse engineering the whole process from scratch from the code.